lochness/mindlamp/__init__.py

```diff
@@ -152,26 +153,36 @@ def sync(Lochness: 'lochness.config',
                 if data_name == 'activity' else get_sensor_events_lamp
 
             # do not re-download already transferred & removed data
-            if is_transferred_and_removed(Lochness, dst):
-                continue
-
-            # confirm it's a new file
-            if Path(dst).is_file():
-                checksum_file = dst.parent / f'.check_sum_{dst.name}'
+            # if is_transferred_and_removed(Lochness, dst):
+                # continue
+
+            prev_file_sha256 = ''  # set previous sha as empty
+            checksum_file = dst.parent / f'.check_sum_{dst.name}'
+            if days_from_ct >= 2 and Path(dst).is_file():
+                # if the days_from_ct is more than two days, the mindlmap data
+                # on the mindlamp server should not change, thus no need to
+                # re-download data for checking checksum.
                 if checksum_file.is_file():
-                    logger.debug(f'{data_name} data exist for {date_str} '
-                                 '- not downloading')
+                    logger.debug(f'{data_name} data has been downloaded for '
+                                 f'{date_str} - skip downloading')
                     continue
 
-                prev_file_sha256 = get_sha(dst)
-            else:
-                prev_file_sha256 = ''
+            elif days_from_ct < 2 and Path(dst).is_file():
+                # potentially within 24 hours from the data acquisition, thus
+                # check if there is any changes in the data on the source
```

Comment:
Suggestion:

                # potentially within 24 hours from the data acquisition, data may change
                # so check if there is any changes in the data on the source

Reply:
0ebf9f49c5fdc035c8dd4028341049a5f9c1931c